### PR TITLE
[feat] Allow use of mime-types in file router config

### PR DESCRIPTION
**Pull request by markflorkowski**
NOTE: Requires corresponding infra changes. 

To Do:
- [x] Infra changes 
- [x] Test to make sure this works and types are 
good

### CODE REVIEW

File: packages/uploadthing/src/mime-types/index.ts [hunk 1]
```diff
@@ -71,8 +71,8 @@ function populateMaps(
   // source preference (least -> most)
   const preference = ["nginx", "apache", undefined, "iana"];
 
-  Object.keys(db).forEach(function forEachMimeType(type) {
-    const mime = db[type];
+  Object.keys(mimeDB).forEach(function forEachMimeType(type) {
+    const mime = mimeDB[type as MimeType] as unknown as MimeObject;
```
Comment (juliusmarminge) [inline packages/uploadthing/src/mime-types/index.ts:75]: these type casts shouldn't be necessary?

File: packages/uploadthing/src/mime-types/index.ts [hunk 2]
```diff
@@ -87,7 +87,10 @@ function populateMaps(
       const extension = exts[i];
 
       if (types[extension]) {
-        const from = preference.indexOf(db[types[extension]].source);
+        const from = preference.indexOf(
+          (mimeDB[types[extension] as MimeType] as unknown as MimeObject)
```
Comment (juliusmarminge) [inline packages/uploadthing/src/mime-types/index.ts:91]: same as above - is the `mimeDB` not typed?

File: packages/uploadthing/src/mime-types/index.ts (hunk 1 above)
Comment (markflorkowski) [reply packages/uploadthing/src/mime-types/index.ts:75]: It got very angry when I didn't have them. Will play with it some more though

File: packages/uploadthing/src/utils.ts [hunk 3]
```diff
@@ -76,6 +81,12 @@ export const getTypeFromFileName = (
     );
   }
 
+  if (allowedTypes.some((type) => type.includes("/"))) {
```
Comment (t3dotgg) [inline packages/uploadthing/src/utils.ts:84]: Would be nice to comment what these cases are for future reference

File: packages/uploadthing/src/types.ts [hunk 4]
```diff
@@ -36,13 +44,17 @@ type RouteConfig = {
...
+>;
 
 type PartialRouteConfig = Partial<
-  Record<AllowedFileType, Partial<RouteConfig>>
+  Record<AllowedFileType | MimeType, Partial<RouteConfig>>
 >;
 
-export type FileRouterInputConfig = AllowedFileType[] | PartialRouteConfig;
+export type FileRouterInputConfig =
+  | (AllowedFileType | MimeType)[]
```
Comment (t3dotgg) [inline packages/uploadthing/src/types.ts:56]: We should prolly define this as like `FileRouterInputKey` and re-use it

**Review by t3dotgg — APPROVED** [review]

**Pull request merged**